streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
@mjsax - leave it as it is. A `Dead` thread isn't queryable. So that would be a bit confusing